core/vm/contract.go

```diff
@@ -84,10 +81,44 @@ func NewContract(caller ContractRef, object ContractRef, value *big.Int, gas uin
 	return c
 }
 
+func (c *Contract) validJumpdest(dest *big.Int) bool {
+	udest := dest.Uint64()
+	// PC cannot go beyond len(code) and certainly can't be bigger than 63bits.
+	// Don't bother checking for JUMPDEST in that case.
+	if dest.BitLen() >= 63 || udest >= uint64(len(c.Code)) {
+		return false
+	}
+	// Only JUMPDESTs allowed for destinations
+	if OpCode(c.Code[udest]) != JUMPDEST {
+		return false
+	}
+	var analysis bitvec
+	// Do we have a contract hash already?
+	if c.CodeHash != emptyCodeHash {
```

Comment:
This is not correct AFAIK. It should be `common.Hash{}`. The `emptyCodeHash` is `Keccak256("")`.